internal/setup/setup_test.go

```diff
@@ -1648,6 +1649,45 @@ func runParseReleaseTests(c *C, tests []setupTest) {
 	}
 }
 
+func (s *S) TestPackageMarshalYAML(c *C) {
+	for _, test := range setupTests {
+		c.Logf("Summary: %s", test.summary)
+
+		if test.relerror == "" || test.release == nil {
+			continue
+		}
+
+		data, ok := test.input["chisel.yaml"]
+		if !ok {
+			data = string(defaultChiselYaml)
+		}
+
+		dir := c.MkDir()
+		// Write chisel.yaml.
+		fpath := filepath.Join(dir, "chisel.yaml")
+		err := os.MkdirAll(filepath.Dir(fpath), 0755)
+		c.Assert(err, IsNil)
+		err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+		c.Assert(err, IsNil)
+		// Write the packages YAML.
+		for _, pkg := range test.release.Packages {
+			fpath = filepath.Join(dir, pkg.Path)
+			err = os.MkdirAll(filepath.Dir(fpath), 0755)
```

Comment:
Isn't the path for all packages a single directory? Why re-making it every time?

Reply:
The packages in `setupTests`are all so far inside the `slices/mydir` directory, yes. But they _can_ be in different directories too, that's why we are re-making it every time.

Reply:
So what you are saying is that currently all tests live in the same directory so there is no need for `MkdirAll`, however somebody might introduce a test that lives in another subfolder (i.e. currently the yaml files can live anywhere, the only requirement is that the first subdir is named `slices`). This setup matches the one in `slicer_test.go` as well, so in my opinion it is better to keep it the way it is to avoid someone writing a test in the future and it failing unexpectedly.

Reply:
Sounds good.